src/vim-qflist-json-error-reporter.cpp

```diff
@@ -120,17 +120,25 @@ vim_qflist_json_error_formatter::vim_qflist_json_error_formatter(
 
 void vim_qflist_json_error_formatter::write_before_message(
     severity sev, const source_code_span &origin) {
-  if (sev == severity::note) {
+  std::string_view severity_value{};
+  switch (sev) {
+  case severity::note:
     // Don't write notes. Only write the main message.
     return;
-  }
+  case severity::error:
+    severity_value = "E";
+    break;
+  case severity::warning:
+    severity_value = "W";
+    break;
+  }  // TODO: handel undifined serverity case.
```

Comment:
What undefined severity cases is this comment talking about?

I'd delete this comment. I don't think there's anything to do here.